### ISSUE TITLE
get_F labeling error when fleets of type 3

### ISSUE THREAD

**Issue by CassidyPeterson-NOAA** [nmfs-fish-tools/SSMSE#168]
get_F() function mislabels fleets in init_F object when fleet type 3 are removed from list of fleetnames input into the function. 

See script where function is called in create_OM(): 
```
 F_list <- SSMSE:::get_F(timeseries = outlist[["timeseries"]], 
                                      fleetnames = dat[["fleetinfo"]][dat[["fleetinfo"]][["type"]] %in%  c(1, 2), "fleetname"])
```
Note that fleetnames subsets only fleets of type 1 and 2. 

See get_F() code below, where the error and fix (that worked for me) is annotated in comments: 

```
> SSMSE:::get_F
function (timeseries, fleetnames) 
{
    assertive.types::assert_is_data.frame(timeseries)
    F_col_ind <- grep("^F:_\\d+$", colnames(timeseries))
    nfleets <- length(F_col_ind)
    assertive.properties::assert_is_of_length(fleetnames, nfleets)
    other_col_ind <- which(colnames(timeseries) %in% c("Yr", 
        "Era", "Seas"))
    F_df <- timeseries[, c(other_col_ind, F_col_ind)]
    F_df <- tidyr::gather(F_df, key = "tmp_Fleet", value = "F", 
        grep("^F:_\\d+$", colnames(F_df)))
    F_df <- tidyr::separate(F_df, col = "tmp_Fleet", into = c(NA, 
        "Fleet"), sep = ":_", convert = TRUE)
    F_rate <- F_df[F_df[["F"]] > 0 & F_df[["Era"]] == "TIME", 
        setdiff(colnames(F_df), c("Era"))]
    if (all(colnames(F_rate) == c("Yr", "Seas", "Fleet", "F"))) {
        colnames(F_rate) <- c("year", "seas", "fleet", "F")
    }
    else {
        stop("Column names not in the correct order.")
    }
    F_rate <- F_rate[order(F_rate[, "fleet"], F_rate[, "year"], 
        F_rate[, "seas"]), ]
    F_rate[["name"]] <- paste0("F_fleet_", F_rate[["fleet"]], 
        "_YR_", F_rate[["year"]], "_s_", F_rate[["seas"]])
    if (nrow(F_rate) == 0) {
        F_rate <- NULL
    }
    init_F <- F_df[F_df[["F"]] > 0 & F_df[["Era"]] == "INIT", 
        c("Seas", "Fleet", "F")]
    if (nrow(init_F) == 0) {
        init_F <- NULL
    }
    else {
        init_F <- init_F[order(init_F[, "Fleet"], init_F[, "Seas"]), 
            ]

         **## ERROR IN NEXT LINE ###
        # fleetnames_df <- data.frame(Fleet = seq_along(fleetnames), 
         #   fleetname = fleetnames)                                                       # Error in CP code
       ## PROPOSED CP CORRECTION ###
         fleetnumbers = which(dat$fleetinfo$fleetname %in% fleetnames) #CP ADDITION ###
         fleetnames_df<- data.frame(Fleet=fleetnumbers, fleetname=fleetnames) #CP FIX ###**

        init_F <- merge(init_F, fleetnames_df)
        init_F_names <- paste0("InitF_seas_", init_F[["Seas"]], 
            "_flt_", init_F[["Fleet"]], init_F[["fleetname"]])
        init_F <- init_F[, "F", drop = TRUE]
        names(init_F) <- init_F_names
    }
    F_rate_fcast <- F_df[F_df[["F"]] > 0 & F_df[["Era"]] == "FORE", 
        setdiff(colnames(F_df), c("Era"))]
    if (all(colnames(F_rate_fcast) == c("Yr", "Seas", "Fleet", 
        "F"))) {
        colnames(F_rate_fcast) <- c("year", "seas", "fleet", 
            "F")
    }
    else {
        stop("Column names not in the correct order.")
    }
    if (nrow(F_rate_fcast) == 0) {
        F_rate_fcast <- NULL
    }
    else {
        F_rate_fcast <- F_rate_fcast[order(F_rate_fcast[, "fleet"], 
            F_rate_fcast[, "year"], F_rate_fcast[, "seas"]), 
            ]
        F_rate_fcast[["name"]] <- paste0("F_fleet_", F_rate_fcast[["fleet"]], 
            "_YR_", F_rate_fcast[["year"]], "_s_", F_rate_fcast[["seas"]])
    }
    F_list <- list(F_df = F_df, F_rate = F_rate, init_F = init_F, 
        F_rate_fcast = F_rate_fcast)
}
<bytecode: 0x000002410378b7d0>
<environment: namespace:SSMSE>
```

Note new object fleetnumbers accounts for any missing fleets of type = 3 that were removed from fleetnames object.

**Comment by k-doering-NOAA** [comment]
Thanks for catching this @CassidyPeterson-NOAA! I just got back from leave and can help solving this within the next week or two!

**Comment by k-doering-NOAA** [comment]
Thanks @CassidyPeterson-NOAA for pushing a fix to this in #174 !
